lang/py/src/avro/io.py

```diff
@@ -119,11 +120,27 @@ def validate(expected_schema, datum):
       return isinstance(datum, Decimal)
     return isinstance(datum, str)
   elif schema_type == 'int':
-    return ((isinstance(datum, int) or isinstance(datum, long)) 
-            and INT_MIN_VALUE <= datum <= INT_MAX_VALUE)
+    if hasattr(expected_schema, 'logical_type'):
```

Comment:
You will have to edit line #117 cause you don't want to be checking this 2 times.